File: src/test/java/pages/geographypage/GeographyNamingPage.java
package pages.geographypage;

import com.microsoft.playwright.Page;
import pages.base.BasePage;

/**
 * Geography Naming Page - Geography Naming Approval
 * Screen: Geography Naming
 */
public class GeographyNamingPage extends BasePage {

    private static final String GEO_NAMING_HEADER = "//h3[text()='Geography Naming']";
    private static final String APPROVE_BTN_1 = "//button[text()=' Approve ']";
    private static final String DOWNLOAD_BTN = "//button[@title='Download Data as Excel']";
    private static final String SAVE_BTN = "//*[text()=' Save ']";

    private static final String SUCCESS_MESSAGE = "Geography Naming successfully approved and moved to next stage.";

    public GeographyNamingPage(Page page) {
        super(page);
    }

    // ============= Screen Validation =============

    public boolean isNamingScreenDisplayed() {
        return isElementVisible(GEO_NAMING_HEADER);
    }

    public boolean isOnNamingScreen() {
        return isOnScreenByURL("geography-naming");
    }

    // ============= Actions =============

    public void clickApprove() {
        clickElement(APPROVE_BTN_1);
    }

    public void clickApproveSecond() {
        page.locator(APPROVE_BTN_1).nth(1).click();
    }

    public void clickSave() {
        clickElement(SAVE_BTN);
    }

    public void clickDownload() {
        clickElement(DOWNLOAD_BTN);
    }

    public void approveNamingWorkflow() {
        clickApprove();
        clickApproveSecond();
    }

    // ============= Verification =============

    public boolean verifyApprovalSuccess() {
        return verifySuccessMessage(SUCCESS_MESSAGE);
    }

    // ============= Profile Data Validation =============

    public void validateProfileData(java.util.Map<String, String> data) {
        // Validates that profile data section is visible on Naming screen
        validateProfileDataIsVisible(data);
    }
}

File: pages/geographyNamingPage.ts
import { Page } from '@playwright/test';
import { BasePage } from './basePage';

export class GeographyNamingPage extends BasePage {
  readonly approveButton;
  readonly confirmApproveButton;
  readonly header;

  constructor(page: Page) {
    super(page);
    this.approveButton = page.locator("//h3[normalize-space()='Geography Naming']//following::button[normalize-space()='Approve']");
    this.confirmApproveButton = page.locator("//h5[normalize-space()='Approve Geography Naming']//following::button[@type='button'][normalize-space()='Approve']");
    this.header = page.locator('//h3[contains(text(),"Geography Naming")]');
  }

  async isGeographyNamingDisplayed(): Promise<boolean> {
    await this.waitForLoaderToDisappear();
    await this.approveButton.waitFor({ state: 'visible', timeout: 10000 });
    return this.header.isVisible();
  }

  async clickApprove(): Promise<void> {
    await this.waitForLoaderToDisappear();
    await this.approveButton.waitFor({ state: 'visible', timeout: 10000 });
    await this.approveButton.click();
  }

  async clickConfirm(): Promise<void> {
    await this.waitForLoaderToDisappear();
    await this.confirmApproveButton.waitFor({ state: 'visible', timeout: 10000 });
    await this.confirmApproveButton.click();
  }

  async verifyApprovalSuccess(message: string): Promise<boolean> {
     const toast = this.page.locator('body').getByText(message, { exact: false });
    await toast.waitFor({ state: 'visible', timeout: 10000 });
    await this.waitForLoaderToDisappear();
    return await toast.isVisible();
  }

  async validateProfileData(): Promise<void> {
    await this.page.getByText(/Profile Data/i).first().waitFor({ state: 'visible' });
  }
}
